caps/management/commands/process_scottish_reporting_data.py

```diff
@@ -0,0 +1,428 @@
+            "Savings figures are estimated or actual",
+        ]
+
+        measurement = "Savings figures are estimated or actual"
+        for col in measurement_columns:
+            if col in targets_df.columns:
+                measurement = col
+
+        for col in [
+            "Primary fuel/emission source saved",
+            measurement,
+            "First full year of CO2e savings",
+        ]:
+            targets_df[col] = targets_df[col].replace(
+                "Please select from drop down box", ""
+            )
+
+        for index, row in targets_df.iterrows():
+            point_type = row["Project name"].strip()
+            point_data = row["Estimated carbon savings per year (tCO2e/annum)"]
+            funding_source = row["Funding source"]
+            if type(funding_source) == str:
+                funding_source = funding_source.strip()
+            self.make_data_point(
+                point_type="project",
+                point_data=point_data,
+                sub_type=point_type,
+                lifetime=row["Project lifetime (years)"],
+                cost=row["Capital cost (£)"],
+                funding_source=funding_source,
+                emission_source=row["Primary fuel/emission source saved"],
+                annual_cost=row["Operational cost (£/annum)"],
+                annual_savings=row["Estimated costs savings (£/annum)"],
+                measurement=row[measurement],
+                savings_start=row["First full year of CO2e savings"],
+                comments=row["Comments"],
+            )
+
+        return 1
+
+    def extract_generation_from_sheet(self, df):
+        if len(df.columns) == 3:
+            col = self.get_description_column()
+            titles = self.sheet.iloc[:, col]
+            start_count = 0
+            end_count = 0
+            for index, item in titles.items():
+                if (
+                    type(item) == str
+                    and re.match(self.subsets["generation"]["start_text"], item)
+                    is not None
+                ):
+                    start_count = index + 1
+                if start_count > 0 and index > start_count and pd.isna(item):
+                    end_count = index
+                    break
+
+            if end_count > start_count:
+                row_range = self.sheet.iloc[start_count:end_count, col:]
+                row_range = row_range.dropna(axis="columns", how="all")
+                df = row_range.iloc[1:]
+
+                columns = row_range.iloc[0, 1:].values
+                columns = np.insert(columns, 0, "Technology")
+                df.columns = columns
+
+                data = {}
+                for index, row in df.iterrows():
+                    if row["Technology"] == "Please select from drop down box":
+                        break
+
+                    consumed = 0
+                    exported = 0
+                    for v in row["Total consumed by the body (kWh)"].values:
```

Comment:
Could you add a comment here explaining why there are multiple values and you prefer the last?